packages/network-controller/src/NetworkController.test.ts

```diff
@@ -3722,6 +3724,108 @@ describe('NetworkController', () => {
       });
     });
   });
+
+  describe('Network Configurations', () => {
+    it('should add a network configuration', async () => {
+      const messenger = buildMessenger();
+      await withController({ messenger }, async ({ controller }) => {
+        const rpcUrlNetwork = {
+          chainId: undefined,
+          nickname: 'RPC',
+          rpcPrefs: undefined,
+          rpcUrl: 'rpc_url',
+          ticker: 'RPC',
+        };
+        const localhostNetwork = {
+          chainId: undefined,
+          nickname: undefined,
+          rpcPrefs: undefined,
+          rpcUrl: 'http://localhost:8545',
+          ticker: 'LOCAL',
+        };
+        controller.upsertNetworkConfiguration(rpcUrlNetwork);
```

Comment:
It looks like we are testing multiple pieces of behavior here. What do we want to test here? If we are testing that when `upsertNetworkConfiguration` is used then a new item gets added to `networkConfigurations`, I wonder if we only need to call `upsertNetworkConfiguration` once here. If there are other nuances we want to test, then I wonder if we want to put those in new tests. What do you think?

Reply:
Yeah for sure. Will clean these up some! I kindof just copied over the tests from the PreferenceController versions. But now would be a very reasonable time to tidy them up!